src/Compilers/CSharp/Test/Semantic/Semantics/TypeParameterCreationTests.cs

```diff
@@ -0,0 +1,272 @@
+    }
+
+    [Theory]
+    [InlineData("new()")]
+    [InlineData("class, new()")]
+    [InlineData("struct")]
+    public void CreateGenericTypeParameterObject_ImplicitCreation(string constraint)
+    {
+        var source = $$"""
+            class C
+            {
+                void M<T>() where T : {{constraint}}
+                {
+                    T t = new();
+                }
+            }
+            """;
+
+        var comp = CreateCompilation(source);
+        comp.VerifyDiagnostics();
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+
+        var objectCreationNode = tree
+            .GetCompilationUnitRoot()
+            .DescendantNodes()
+            .OfType<ImplicitObjectCreationExpressionSyntax>()
+            .Single();
+
+        var typeInfo = model.GetTypeInfo(objectCreationNode);
+        Assert.Equal("T", typeInfo.Type.ToTestDisplayString());
+        Assert.Equal("T", typeInfo.ConvertedType.ToTestDisplayString());
+
+        var symbolInfo = model.GetSymbolInfo(objectCreationNode);
+        Assert.True(symbolInfo.IsEmpty);
+    }
+
+    [Theory]
+    [InlineData("")]
+    [InlineData("where T : class")]
+    public void CreateGenericTypeParameterObject_ExplicitCreation_ErrorRecovery_NoNewConstraint(string constraintClause)
+    {
+        var source = $$"""
+            class C
+            {
+                void M<T>() {{constraintClause}}
+                {
+                    T t = new T();
+                }
+            }
+            """;
+
+        var comp = CreateCompilation(source);
+        comp.VerifyDiagnostics(
+            // (5,15): error CS0304: Cannot create an instance of the variable type 'T' because it does not have the new() constraint
+            //         T t = new T();
+            Diagnostic(ErrorCode.ERR_NoNewTyvar, "new T()").WithArguments("T").WithLocation(5, 15));
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+
+        var objectCreationNode = tree
+            .GetCompilationUnitRoot()
+            .DescendantNodes()
+            .OfType<ObjectCreationExpressionSyntax>()
+            .Single();
+
+        var typeInfo = model.GetTypeInfo(objectCreationNode);
+        Assert.Equal("T", typeInfo.Type.ToTestDisplayString());
+        Assert.Equal("T", typeInfo.ConvertedType.ToTestDisplayString());
+
+        var symbolInfo = model.GetSymbolInfo(objectCreationNode);
+        Assert.True(symbolInfo.IsEmpty);
+
+        var typeNode = objectCreationNode.Type;
+
+        typeInfo = model.GetTypeInfo(typeNode);
+        Assert.Null(typeInfo.Type);
+        Assert.Null(typeInfo.ConvertedType);
+
+        symbolInfo = model.GetSymbolInfo(typeNode);
+        Assert.Null(symbolInfo.Symbol);
+        Assert.Equal(CandidateReason.NotCreatable, symbolInfo.CandidateReason);
+        Assert.Collection(symbolInfo.CandidateSymbols,
+            s => Assert.Equal("T", s.ToTestDisplayString()));
```

Comment:
@333fred I didn't change this scenario because it fully matches behavior with normal types: e.g. if we have some `new X()` expression and get symbol info from just `X` we will get symbol `X` if the expression itself is correct or `X` as a candidate symbol with candidate reason `NotCreatable` if expression has errors (e.g. if `X` is a `static` or an `abstract` class). With type parameters we get the same behavior: `T` if `new T()` expression is legal and candidate `T` with reason `NotCreatable` if parent object creation expression is an error. Does this look correct to you?